spot_wrapper/wrapper.py

```diff
@@ -975,21 +977,39 @@ def releaseEStop(self) -> None:
         """Stop eStop keepalive"""
         if self._estop_keepalive:
             self._estop_keepalive.stop()
+            self._estop_keepalive.shutdown()
             self._estop_keepalive = None
             self._estop_endpoint = None
 
-    def getLease(self) -> None:
+    def takeLease(self) -> typing.Tuple[bool, typing.Optional[Lease]]:
+        """Take lease for the robot, which may have been taken already."""
+        lease = self._lease_client.take()
+        have_new_lease = (self._lease is None and lease is not None) or (
+            str(lease.lease_proto) != str(self._lease.lease_proto)
+        )
+        if have_new_lease:
+            if self._lease_keepalive is not None:
+                self._lease_keepalive.shutdown()
+            self._lease_keepalive = LeaseKeepAlive(self._lease_client)
+            self._lease = lease
+        return have_new_lease, self._lease
+
+    def getLease(self) -> typing.Optional[Lease]:
         """Get a lease for the robot and keep the lease alive automatically."""
-        if self._use_take_lease:
-            self._lease = self._lease_client.take()
-        else:
-            self._lease = self._lease_client.acquire()
+        if not self._lease:
+            if self._use_take_lease:
+                self._lease = self._lease_client.take()
+            else:
+                self._lease = self._lease_client.acquire()
```

Comment:
So I think we probably still need to call `take` or `acquire` because leases can be overridden.  My guess is that as written here once something takes control from the driver, the driver won't be able to take it back.

Reply:
Yeap, you are right. I misunderstood the `claim` implementation. Fixed in https://github.com/bdaiinstitute/spot_wrapper/pull/92/commits/4463731cba1330827f727799229c1d3c2f452b40.